tensorflow_datasets/text/tasks.py

```diff
@@ -0,0 +1,162 @@
+
+def _test_data_filenames(tmp_dir):
+    return tf.io.gfile.glob(os.path.join(tmp_dir, _TEST_FILE_FORMAT))
+
+class tasks(tfds.core.GeneratorBasedBuilder):
+    """ Tasks dataset combining useful information of training and test data """
+    BUILDER_CONFIGS = [
+        TasksConfig(
+            name="plain_text",
+            version="0.0.1",
+            description="Plain text",
+        ),
+        TasksConfig(
+            name="bytes",
+            version="0.0.1",
+            description=("Uses byte-level text encoding with "
+                         "`tfds.features.text.ByteTextEncoder`"),
+            text_encoder_config=tfds.features.text.TextEncoderConfig(
+                encoder=tfds.features.text.ByteTextEncoder()),
+        ),
+        TasksConfig(
+            name="subwords8k",
+            version="0.0.2",
+            description=("Uses `tfds.features.text.SubwordTextEncoder` with 8k "
+                         "vocab size"),
+            text_encoder_config=tfds.features.text.TextEncoderConfig(
+                encoder_cls=tfds.features.text.SubwordTextEncoder,
+                vocab_size=2**13),
+        ),
+        TasksConfig(
+            name="subwords32k",
+            version="0.0.2",
+            description=("Uses `tfds.features.text.SubwordTextEncoder` with "
+                         "32k vocab size"),
+            text_encoder_config=tfds.features.text.TextEncoderConfig(
+                encoder_cls=tfds.features.text.SubwordTextEncoder,
+                vocab_size=2**15),
+        ),
+    ]
+
+
+    def _info(self):
+        # Specifies tfds.core.DatasetInfo object
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict({
+                "text":tfds.features.Text(
+                    encoder_config=self.builder_config.text_encoder_config),
+                        }),
+            supervised_keys=("text", "text"),
+            urls=["http://www.thespermwhale.com/jaseweston/babi/"],
+            citation=_CITATION
+                )
+
+    def _vocab_text_gen(self, training_files):
+        for ex in self._generate_examples(training_files):
+            yield ex["text"]
+
+    def _split_generators(self, dl_manager):
+        tasks_path = dl_manager.download_and_extract(_DOWNLOAD_URL)
+        train_files = _train_data_filenames(tasks_path)
+        test_files = _test_data_filenames(tasks_path)
+        # Generate vocabulary from training data if SubwordTextEncoder configured
+        self.info.features["text"].maybe_build_from_corpus(
+            self._vocab_text_gen(train_files))
+
+        return [
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TRAIN,
+                num_shards=10,
+                gen_kwargs={"files": train_files}),
+            tfds.core.SplitGenerator(
+                name=tfds.Split.TEST,
+                num_shards=10,
```

Comment:
how large is this dataset? we usually expect 2GB/shard

Reply:
ohh sorry sir i did not know that

Reply:
this dataset is 20 mb so can suggest me some shards

Reply:
if this is only 20 MB, then 1 shard should be enough

Reply:
ok sir thanks can you help me in resolving this error
I am not able to understand where this test case fails 

self = <tensorflow_datasets.text.BablTasks_test.BablTasksTest testMethod=test_download_and_prepare_as_dataset>, args = (), kwargs = {}

    def decorated(self, *args, **kwargs):
      """Run the decorated test method."""
      if not tf.executing_eagerly():
        raise ValueError("Must be executing eagerly when using the "
                         "run_in_graph_and_eager_modes decorator.")
    
      # Run eager block
>     f(self, *args, **kwargs)

../testing/test_utils.py:170: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../testing/dataset_builder_testing.py:193: in test_download_and_prepare_as_dataset
    self._download_and_prepare_as_dataset(builder)
../testing/dataset_builder_testing.py:213: in _download_and_prepare_as_dataset
    self._assertAsDataset(builder)
../testing/dataset_builder_testing.py:237: in _assertAsDataset
    self.assertLen(examples, expected_examples_number)
/home/shashank/miniconda3/lib/python3.7/site-packages/absl/testing/absltest.py:784: in assertLen
    container_repr, len(container), expected_len), msg)
/home/shashank/miniconda3/lib/python3.7/site-packages/absl/testing/absltest.py:1609: in fail
    return super(TestCase, self).fail(self._formatMessage(prefix, msg))
E   AssertionError: [] has length of 0, expected 3.